#CNV-43159_how_to_detect (#93637)
openshift/openshift-docs: modules/virt-deprecated-tasks.adoc

// Module included in the following assemblies:
//
// * virt/virtual_machines/virt-managing-vms-openshift-pipelines.adoc

:_mod-docs-content-type: REFERENCE
[id="virt-deprecated-tasks_{context}"]
= Removing deprecated or unused resources

> * The tekton-tasks-operator (TTO) has been deprecated and removed in {VirtProductName} 4.17. You can clean up any remaining resources associated with the operator and its pipelines by running the following command.

Reviewer: maybe don't mention pipelines explicitly.. 
resources associated with the operator should cover all

> * If you no longer use Tekton (TTO) in your environment after upgrading to OpenShift Container Platform 4.14 or later, you can remove any remaining Tekton-related resources that were previously managed by the ssp-operator.

Reviewer 2: - tekton-task-operator was deprecated in 4.14 and remained in the code and supported until it was removed from the code in v4.17.
- Please do not abbreviate Tekton as TTO. We never used the acronym in our documentation. The documentation always spelled-out tekton-tasks-operator.
- Best practice: Keep sentences less than 30 words.
- Best practice, when starting a sentence with if, then include the "then" after the comma.

Recommend re-writing:

The tekton-tasks-operator was deprecated and removed from OpenShift Virtualization 4.17. You can remove any remaining resources that were used by the tekton-tasks-operator and pipelines by running the following command.

.Procedure

> . The following command removes any remaining tekton-related resources from the cluster:

Reviewer 3: ```suggestion
. To remove any remaining Tekton-related resources from the cluster, run the following command:
```

Reviewer 3: See this style resource regarding commands: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command

> . Use the following command to remove any leftover Tekton-related resources from the cluster:

Reviewer 2: Recommend:
The following command removes any remaining Tekton-related resources from the cluster:

[source,terminal]
----
oc delete clusterroles,rolebindings,serviceaccounts,configmaps,pipelines,tasks \
  --selector 'app.kubernetes.io/managed-by=ssp-operator' \
  --selector 'app.kubernetes.io/component in (tektonPipelines,tektonTasks)' \
  --selector 'app.kubernetes.io/name in (tekton-pipelines,tekton-tasks)' \
  --ignore-not-found \
  --all-namespaces
----

> * If the Tekton tasks operator custom resource definitions (CRDs) have already been removed, the command may return an error. You can safely ignore this, as all other matching resources will still be deleted.

Reviewer 3: ```suggestion
* If the Tekton tasks operator custom resource definitions (CRDs) have already been removed, the command might return an error. You can safely ignore the error, as all other matching resources are still deleted.
```

Reviewer 3: The use of may vs. might is covered in the IBM Style Guide [Word usage](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m) section.

The use of tense is covered here in the [IBM Style Guide](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#tense).

> . If the Tekton custom resource definitions (CRDs) have already been removed, the command may return an error. You can safely ignore this, as all other matching resources will still be deleted.

Reviewer: both can't be 1

> . If the Tekton custom resource definitions (CRDs) have already been removed, the command might return an error. You can safely ignore this error. All other matching resources will still be deleted.

Reviewer 2: - If/then
- remove "still" (minimalism)
- change leftover to remaining

Recommend:
If the Tekton custom resource definitions (CRDs) have already been removed, then the command might return an error. You can safely ignore this error. All other matching resources will be deleted.
